src/main/prometheus/provider.ts

```diff
@@ -0,0 +1,28 @@
+export type PrometheusQuery = {
+  [key: string]: string;
```

Comment:
This is not exactly typesafe. Could we list here all known keys so that we can be sure each provider implements all required queries?

Reply:
Refactored the implementation to be type safer.